inc/class-tmsc-linked-taxonomy-posts.php

```diff
@@ -0,0 +1,358 @@
+
+						// If filter hook above has been applied, then write a child term.
+						$parent_term = 0;
+						if ( ! empty( $this->parent_term ) ) {
+							$parent_term = $this->parent_term->term_id;
+						}
+
+						$args = array(
+							'slug' => $post->post_name,
+							'parent' => ( ! empty( $parent_term ) ) ? $parent_term : 0,
+						);
+						$term = wp_insert_term( $post->post_title, $taxonomy, $args );
+
+						if ( ! empty( $term ) && ! is_wp_error( $term ) ) {
+							update_term_meta( $term['term_id'], $this->linked_term_meta_key, $post_id );
+							update_post_meta( $post_id, $this->linked_post_meta_key, $term['term_id'] );
+							update_post_meta( $post_id, $this->linked_taxonomy_meta_key, $taxonomy );
+							$term = get_term( $term['term_id'], $taxonomy );
+						}
+					} else {
+						$term = $this->get_linked_term( $post->post_parent );
+						// Set the meta data for the children.
+						update_term_meta( $term->term_id, $this->linked_term_meta_key, $post->post_parent );
+						update_post_meta( $post_id, $this->linked_post_meta_key, $term->term_id );
+						update_post_meta( $post_id, $this->linked_taxonomy_meta_key, $taxonomy );
+					}
+				}
+			}
+		} // End if().
+
+		$this->clear_linked_cache( $post_id );
+		if ( ! empty( $term ) && ! is_wp_error( $term ) ) {
+			return $term;
+		}
+	}
+
+
+	/**
+	 * Run through a series of checks to grab a query arg.
+	 */
+	public function get_query_arg( $name ) {
+		global $taxnow, $typenow;
+		$value = '';
+		$url_args = wp_parse_args( wp_parse_url( wp_get_referer(), PHP_URL_QUERY ) );
+
+		switch( true ) {
+			case ( ! empty( get_query_var( $name ) ) ) :
+				$value = get_query_var( $name );
+				break;
+			// If the query arg isn't set. Check POST and GET requests.
+			case ( ! empty( $_POST[ $name ] ) ) :
+				$value = sanitize_text_field(  $_POST[ $name ] );
+				break;
+			case ( ! empty( $_GET[ $name ] ) ) :
+				$value = sanitize_text_field(  $_GET[ $name ] );
+				break;
+			case ( 'post_type' === $name && ! empty( $typenow ) ) :
+				$value = $typenow;
+				break;
+			case ( 'taxonomy' === $name && ! empty( $taxnow ) ) :
+				$value = $taxnow;
+				break;
+			case ( ! empty( $url_args[ $name ] ) ) :
+				$value = $url_args[ $name ];
+				break;
+			default :
+				$value = '';
+		}
+		return $value;
+	}
+
+	/**
+	 * Delete a linked taxonomy when it's post is deleted.
+	 * @param int $post_id.
+	 * @return void.
+	 */
+	public function delete_linked_taxonomy( $post_id ) {
+		$term = $this->get_linked_term( $post_id );
+		if ( ! empty( $term ) ) {
+			wp_delete_term( $term->term_id, $term->taxonomy );
+		}
+		$this->clear_linked_cache( $post_id );
+	}
+
+	/**
+	 * Delete any associated child posts.
+	 * @param int $post_id.
+	 * @return void.
+	 */
+	public function delete_child_posts( $post_id ) {
+		$post = get_post( $post_id );
+		$children = get_posts( array(
+			'post_type' => $post->post_type,
+			'post_parent' => $post->ID,
+		) );
+		foreach( $children as $child ) {
```

Comment:
I would add a check to make sure that $children is an array and not empty before looping through it.